app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,211 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import Recipe, Ingredient
+
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def _create_recipe(self):
+    payload = {
+        'name': 'Pizza',
+        'description': 'Put it in the oven',
+        'ingredients': [
+            {'name': 'dough'},
+            {'name': 'cheese'},
+            {'name': 'tomato'}
+        ]
+    }
+    res = self.client.post(RECIPES_URL, data=payload, format='json')
+    return res
+
+
+def detail_url(recipe_id):
+    """Return recipe detail URL"""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def search_url(name):
+    """Return recipe search URL"""
+    return reverse('recipe:recipe-list') + f'?name={name}'
+
+
+class RecipeApiTest(TestCase):
+    """Tests for the Recipe API"""
+
+    def setUp(self):
```

Comment:
Picky one: you could better use setUpClass since this would be called only once for the test class and not for each test
https://docs.python.org/3/library/unittest.html#setupclass-and-teardownclass